tests/testthat/test-modspec.R

```diff
@@ -163,9 +149,192 @@ test_that("at options are parsed", {
+$CMT @object pcmt
+'
+  
+  mod <- mcode("foo", code, compile = FALSE)
+  x <- labels(mod)
+  expect_equal(x$param, c("a", "b", "theta1", "theta2", "z"))
+  expect_equal(mod$z, 777)
+  expect_equal(x$omega_labels[[3]], c("A", "B", "C"))
+  expect_equal(x$omega[2], "omega2")
+  d <- nrow(omat(mod))
+  expect_equal(unname(nrow(omat(mod))), c(1,2,3))
+  expect_equal(unname(nrow(smat(mod))), c(1,4,5))
+  expect_equal(x$init, c("gg", "hh", "iii", "t", "u", "v"))
+})
+
+test_that("parse content using low-level handlers - PARAM", {
+  env <- mrgsolve:::parse_env(vector(mode = "list", length = 20), project = '.')
+  sup <- suppressMessages  
+  
+  input <- "c(1,2,3)"
+  expect_error(
+    sup(mrgsolve:::PARAM(x = input, as_object = TRUE)), 
+    "code returned the incorrect class: numeric"
+  )
+  input <- "list(a = 1, b = 2)"
+  ans <- mrgsolve:::PARAM(x = input, as_object = TRUE, env = env, pos = 3)
+  expect_is(env$param[[3]], "list")
+  expect_named(env$param[[3]])
+  
+  expect_null(env$param[[8]])  
+  env$ENV$parameters <- list(mm = 1, nn = 2)
+  ans <- mrgsolve:::PARAM(x = input, object = "parameters", env = env, pos = 8)
+  expect_is(env$param[[8]], "list")
+  expect_named(env$param[[8]])
+})
+
+test_that("parse content using low-level handlers - THETA", {
+  env <- mrgsolve:::parse_env(vector(mode = "list", length = 20), project = '.')
+  sup <- suppressMessages  
+  
+  input <- "list(1,2,3)"
+  expect_error(
+    sup(mrgsolve:::THETA(x = input, as_object = TRUE)), 
+    "code returned the incorrect class: list"
+  )
+  input <- "c(3,4,5,6)"
+  ans <- mrgsolve:::THETA(x = input, as_object = TRUE, env = env, pos = 10)
+  expect_is(env$param[[10]], "list")
+  expect_named(env$param[[10]])
+  
+  expect_null(env$param[[2]])  
+  env$ENV$thetas <- c(9,8,7,6,5)
+  ans <- mrgsolve:::THETA(x = "", object = "thetas", env = env, pos = 2)
+  expect_is(env$param[[2]], "list")
+  expect_named(env$param[[2]])
+})
+
+test_that("parse content using low-level handlers - CMT", {
+  env <- mrgsolve:::parse_env(vector(mode = "list", length = 20), project = '.')
+  sup <- suppressMessages  
+  
+  input <- "c(2,2,3)"
+  expect_error(
+    sup(mrgsolve:::CMT(x = input, as_object = TRUE)), 
+    "code returned the incorrect class: numeric"
+  )
+  input <- "letters[1:3]"
+  ans <- mrgsolve:::CMT(x = input, as_object = TRUE, env = env, pos = 8)
+  expect_is(env$init[[8]], "numeric")
+  expect_named(env$init[[8]])
+  
+  expect_null(env$param[[2]])  
+  env$ENV$compartments <- letters[8:12]
+  ans <- mrgsolve:::CMT(x = "", object = "compartments", env = env, pos = 2)
+  expect_is(env$init[[2]], "numeric")
+  expect_named(env$init[[2]])
+})
+
+test_that("parse content using low-level handlers - INIT", {
+  env <- mrgsolve:::parse_env(vector(mode = "list", length = 20), project = '.')
+  sup <- suppressMessages  
+  
+  input <- "c(2,2,3)"
+  expect_error(
+    sup(mrgsolve:::INIT(x = input, as_object = TRUE)), 
+    "code returned the incorrect class: numeric"
+  )
```

Comment:
Should there also be a test for an unnamed list?

Reply:
Thanks; testing for names now.